pallets/rewards/src/lib.rs

```diff
@@ -268,14 +267,18 @@ pub mod pallet {
 
 				Groups::<T>::try_mutate(group_id, |group| {
 					StakeAccounts::<T>::try_mutate(account_id, currency_id, |account| {
-						T::Currency::hold(currency_id.1, account_id, amount)?;
+						if !T::Currency::can_hold(currency_id.1, account_id, amount) {
+							Err(TokenError::NoFunds)?;
+						}
 
 						account.try_apply_rpt_tallies(currency.rpt_tallies())?;
 						account.add_amount(amount, group.reward_per_token())?;
 
 						group.add_amount(amount)?;
 						currency.add_amount(amount)?;
 
+						T::Currency::hold(currency_id.1, account_id, amount)?;
```

Comment:
Maybe leave a comment explaining what `hold` is/does and its application in this context? I guess it's used here to stake `amount`?

Reply:
> I guess it's used here to stake amount?

Yes! It's the core of this pallet, `hold()` is the same as `Currency::reserve()` but for tokens. This change only moves down that line to the bottom.

Reply:
Not sure about the comment here, but maybe it is good to explain this on top of the file as general description of how it works

Reply:
yeah sounds good!